File: user-service/src/main/java/com/acme/usersrv/company/service/CompanyService.java
package com.acme.usersrv.company.service;

import com.acme.usersrv.company.CompanyStatus;
import com.acme.usersrv.company.dto.CompanyDto;
import com.acme.usersrv.company.dto.CompanyFilter;
import com.acme.usersrv.company.dto.FullDetailsCompanyDto;
import com.acme.usersrv.company.dto.RegisterCompanyDto;
import com.acme.usersrv.company.dto.UpdateCompanyDto;
import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.security.access.prepost.PreAuthorize;
import org.springframework.validation.annotation.Validated;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

import javax.validation.Valid;
import javax.validation.constraints.NotNull;
import java.util.Collection;
import java.util.UUID;

@Validated
public interface CompanyService {
    Mono<UUID> register(@Valid RegisterCompanyDto registrationDto);

    @PreAuthorize("hasAuthority('ADMIN')")
    Mono<Page<FullDetailsCompanyDto>> find(CompanyFilter filter, Pageable pageable);

    @PreAuthorize("hasAuthority('ADMIN')")
    Mono<Page<FullDetailsCompanyDto>> findByJooq(CompanyFilter filter, Pageable pageable);

    @PreAuthorize("hasAuthority('ADMIN')")
    Mono<Void> changeStatus(UUID id, @NotNull CompanyStatus status);

    @PreAuthorize("isAuthenticated()")
    Mono<CompanyDto> findById(UUID id);

    @PreAuthorize("hasAnyAuthority('ADMIN', 'COMPANY_OWNER','PP_MANAGER')")
    Mono<FullDetailsCompanyDto> findFullDetailsById(UUID id);

    @PreAuthorize("hasAnyAuthority('ADMIN', 'COMPANY_OWNER')")
    Mono<Void> update(UUID id, @Valid UpdateCompanyDto dto);

    @PreAuthorize("hasAnyAuthority('ADMIN, ACCOUNTANT')")
    Flux<CompanyDto> findNames(Collection<CompanyStatus> status);
}

File: admin-ui/src/common/CompanyService.js
import RestApi, { buildGetFilterParams } from './RestApi';


const CompanyService = {
    register: function (request) {
        return RestApi.post('/user-service/companies', request);
    },
    find: function (filter, pageable, sort) {
        return RestApi.get('/user-service/companies', {
            params: buildGetFilterParams(filter, pageable, sort)
        });
    },
    changeStatus: function (id, request) {
        return RestApi.put(`/user-service/companies/${id}/status`, request);
    },
    findById: function (id) {
        return RestApi.get(`/user-service/companies/${id}`);
    },
    findByIdFullDetails: function (id) {
        return RestApi.get(`/user-service/companies/${id}/full-details`);
    },
    update: function (id, request) {
        return RestApi.put(`/user-service/companies/${id}`, request);
    }
    ,
    findNames: function (statuses) {
        let urlSearchParams = new URLSearchParams();
        if (statuses && statuses.length) {
            statuses.forEach(status => {
                urlSearchParams.append("status", status);
            });
        }
        return RestApi.get(`/user-service/companies/names`, {
            params: urlSearchParams
        });
    }
}

export default CompanyService;